bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetThermoregulationHandler.java

```diff
@@ -399,65 +491,85 @@ protected void handleMessage(BaseOpenMessage msg) {
 
     private void updateModeAndFunction(Thermoregulation tmsg) {
         if (tmsg.getWhat() == null) {
-            logger.warn("updateModeAndFunction() Could not parse Mode or Function from {} (WHAT is null)",
-                    tmsg.getFrameValue());
+            logger.warn("updateModeAndFunction() Could not parse What {} (WHAT is null)", tmsg.getFrameValue());
             return;
         }
-        Thermoregulation.WhatThermo w = Thermoregulation.WhatThermo.fromValue(tmsg.getWhat().value());
-        if (w.getMode() == null) {
+        Thermoregulation.WhatThermo what = tmsg.new WhatThermo(tmsg.getWhat().value());
```

Comment:
I don't know this way to call a constructor. For my information, what does exactly `tmsg.new WhatThermo` ?

Reply:
It's the right way to create a new instance of an inner class. 
Did not know that, but Eclipse highlights a compilation error if i use `new Thermoregulation.WhatThermo()`:

_No enclosing instance of type Thermoregulation is accessible. Must qualify the allocation with an enclosing instance of type Thermoregulation (e.g. x.new A() where x is an instance of Thermoregulation)._